brooklyn-tosca-transformer/src/main/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaNodeToEntityConverter.java

```diff
@@ -355,9 +386,73 @@ protected void applyLifecycle(Map<String, Operation> ops, String opKey, EntitySp
 
     }
 
+    protected void configureRuntimeEnvironment(EntitySpec<?> entitySpec) {
+        if (indexedNodeTemplate.getArtifacts() == null) {
+            return;
+        }
+
+        final Map<String, String> filesToCopy = MutableMap.of();
+        final List<String> preInstallCommands = MutableList.of();
+        final List<String> envCommands = MutableList.of();
+
+        for (final Map.Entry<String, DeploymentArtifact> artifactEntry : indexedNodeTemplate.getArtifacts().entrySet()) {
+            if (artifactEntry.getValue() == null) {
+                continue;
+            }
+            if (!"tosca.artifacts.File".equals(artifactEntry.getValue().getArtifactType())) {
+                continue;
+            }
+
+            final String destRoot = Os.mergePaths("~", artifactEntry.getValue().getArtifactName());
+            final String tempRoot = Os.mergePaths("/tmp", artifactEntry.getValue().getArtifactName());
+
+            preInstallCommands.add("mkdir -p " + destRoot);
+            preInstallCommands.add("mkdir -p " + tempRoot);
+            envCommands.add(String.format("export %s=%s", artifactEntry.getValue().getArtifactName(), destRoot));
+
+            try {
+                Path csarPath = csarFileRepository.getCSAR(artifactEntry.getValue().getArchiveName(), artifactEntry.getValue().getArchiveVersion());
+
+                Path resourcesRootPath = Paths.get(csarPath.getParent().toAbsolutePath().toString(), "expanded", artifactEntry.getKey());
+                Files.walkFileTree(resourcesRootPath, new SimpleFileVisitor<Path>() {
+                    @Override
+                    public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                        String tempDest = Os.mergePaths(tempRoot, file.getFileName().toString());
+                        String finalDest = Os.mergePaths(destRoot, file.getFileName().toString());
+                        filesToCopy.put(file.toAbsolutePath().toString(), tempDest);
```

Comment:
If you give an explicit path such as `~/foo/my.file`, Brooklyn will created a folder called `~` in the home directory. If you give a relative path such as `foo/my.file`, then Brooklyn will copy the files to the install directory, however we need to know the explicit path so that it can be injected into the lifecycle scripts and the install dir is not known at this point